lnwallet/chainfee/estimator.go

```diff
@@ -868,7 +877,22 @@ func (w *WebAPIEstimator) Stop() error {
 //
 // NOTE: This method is part of the Estimator interface.
 func (w *WebAPIEstimator) RelayFeePerKW() SatPerKWeight {
-	return FeePerKwFloor
+	// Get fee estimates now if we don't refresh periodically.
+	if w.noCache {
+		w.updateFeeEstimates()
+	}
+
+	if w.minRelayFeerate == 0 {
+		log.Errorf("No min relay fee rate available, using default %v",
+			FeePerKwFloor)
+
+		return FeePerKwFloor
+	}
```

Comment:
Ah nvm, we are interpreting 0 as unset. May I suggest converting this to an option then and using UnwrapOr?

Reply:
you could set a 0 in the API resp, but it's not allowed, so I guess it's a bit different (whenever I wanna differentiate the case of `unset` vs the default empty value I'd use `Option`). And I wanna log that error tho, don't think `UnwrapOr` can do that?

Reply:
what you could do is: 

in `parseResponse` where you define the response struct, add all default values there. Then only values actually specified by the API source would be overwritten. 

Then error there if after parsing, you are left with a zero. 

That way you allow the api to set to zero but then you log an error there at response parsing time (which is when the error is really happening). 

But if you really never want to allow 0, then this way is probs fine too? no super strong opinion here.

Reply:
> And I wanna log that error tho, don't think UnwrapOr can do that?

Not on its own but you could do this:
```golang
if w.minRelayFeerate.IsNone() {
        log.Errorf("No min relay fee rate available, using default %v",
			FeePerKwFloor)
}
return w.minRelayFeerate.UnwrapOr(FeePerKwFloor)
```

I think we should try and kick the habit of using magic values to indicate invalid states.

Reply:
Cool I updated the check - long story short - tried the option approach, realized we cannot get away with the zero check since we are parsing from a json resp, so it has to happen somewhere - updated the parse method so we don't need to worry about it being zero.